CONTRIBUTING.md

```diff
@@ -166,86 +92,77 @@ conda activate cudf_dev
+- A `build.sh` script is provided in `$CUDF_HOME`. Note that  CMake depends on the `nvcc` executable being on your path or defined in `$CUDACXX` and the libraries will be installed to the location set in `$INSTALL_PREFIX` if set (i.e. `export INSTALL_PREFIX=/install/path`), otherwise to `$CONDA_PREFIX`.
 ```bash
 $ cd $CUDF_HOME
-$ ./build.sh                                                              # To build both C++ and Python cuDF versions with their dependencies
+$./build.sh libcudf             
 ```
-- To build only the C++ component with the script
+- Many other librairies like `dask-cudf` and `cudf-kafka` can be installed with this script. For the complete list of librairies as well as details about the script usage, run the `help` command:
 ```bash
-$ ./build.sh libcudf                                                      # Build only the cuDF C++ components and install them to $INSTALL_PREFIX if set, otherwise $CONDA_PREFIX
+$ cd $CUDF_HOME
+$./build.sh --help            
 ```
 
-- To run tests (Optional):
+### Install and Test cuDF for contributors
+
+The general workflow is to find your contributor profile (C++ or Python, etc...), then follow the corresponding instructions and finally go to the last section about [code formatting](###code-formatting).
+
+#### Any contributor:
+
+This section provides instructions for building and testing C++ libcudf. Thus, C++ libcudf only contributor will be all set after following instructions below:
+ 
 ```bash
-$ make test
+$ cd $CUDF_HOME
+$ # for C++ contributors 
+$./build.sh libcudf tests                        # building C++ cuDF and test components 
+$ make test                                      # running C++ cuDF unit tests
+$ # for other contributors
+$./build.sh libcudf                     
 ```
-- Build the `cudf` python package, in the `python/cudf` folder:
+
+#### Python cudf contributors:
+
+- To build and install in edit/develop mode the `cudf` python package, in the `python/cudf` folder:
 ```bash
 $ cd $CUDF_HOME/python/cudf
 $ python setup.py build_ext --inplace
-$ python setup.py install
+$ python setup.py develop
 ```
 
-- Like the `libcudf` build step above, `build.sh` can also be used to build the `cudf` python package, as shown below:
+- To run `cudf` tests :
 ```bash
-$ cd $CUDF_HOME
-$ ./build.sh cudf
+$ cd $CUDF_HOME/python
+$ py.test -v cudf/cudf/tests                     # run cudf test suite
 ```
 
-- Additionally to build the `dask-cudf` python package, in the `python/dask_cudf` folder:
-```bash
-$ cd $CUDF_HOME/python/dask_cudf
-$ python setup.py install
-```
+#### Python dask-cudf contributors:
 
-- The `build.sh` script can also  be used to build the `dask-cudf` python package, as shown below:
+- To build and install in edit/develop mode the `dask-cudf` python package, in the `python/dask_cudf` folder:
```

Comment:
Note that there is no "build" step for `dask-cudf` as it's a Python-only library :) 

```suggestion
- To install in edit/develop mode the `dask-cudf` python package:
```

Reply:
Thank you for correcting it !